apps/src/dance/ai/DanceAiClient.ts

```diff
@@ -40,66 +42,62 @@ const CachedPalettes: CachedWeightsMapping = UntypedCachedPalettes;
  * }
  */
 export function chooseEffects(
-  emojis: string[],
+  selectedEmojis: string[],
   quality: ChooseEffectsQuality
 ): GeneratedEffect {
   // Obtain final summed output weight based off input received
-  const outputTypes: CachedWeightsMapping[] = [
-    CachedBackgrounds,
-    CachedForegrounds,
-    CachedPalettes,
-  ];
-  const outputWeights: number[][] = outputTypes.map(function (map) {
-    return calculateOutputWeightsVector(emojis, map);
-  });
-
-  // Sort and slice top scoring options, mapped to their output identifiers (e.g. [[0.25, 'squiggles'], ...])
+  const cachedWeightsMappings: {[key: string]: CachedWeightsMapping} = {
+    backgroundEffect: CachedBackgroundEffects,
+    foregroundEffect: CachedForegroundEffects,
+    backgroundColor: CachedPalettes,
+  };
+  const outputWeightsForSelectedEmojis: {[key: string]: number[]} = {};
+  for (const output in cachedWeightsMappings) {
+    outputWeightsForSelectedEmojis[output] = calculateOutputWeightsVector(
+      selectedEmojis,
+      cachedWeightsMappings[output]
+    );
+  }
+
+  // Sort and slice top or bottom scoring options, mapped to their output identifiers (e.g. [[0.25, 'squiggles'], ...])
   const numRandomOptions = 3;
-  const outputOptions: [number, string][][] = outputWeights.map(
-    (weightVector, i) => {
-      const optionsAll = obtainOptions(weightVector, outputTypes[i]);
-      const options =
-        quality === ChooseEffectsQuality.GOOD
-          ? optionsAll.slice(0, numRandomOptions)
-          : optionsAll.slice(-numRandomOptions);
-      return options;
-    }
-  );
-
-  // Clarification of which array correlates to which option.
-  const backgroundOptions: [number, string][] = outputOptions[0];
-  const foregroundOptions: [number, string][] = outputOptions[1];
-  const paletteOptions: [number, string][] = outputOptions[2];
-
-  // Choose random values.
-  const backgroundEffectOption =
-    backgroundOptions[Math.floor(Math.random() * backgroundOptions.length)];
-  const backgroundColorOption =
-    paletteOptions[Math.floor(Math.random() * paletteOptions.length)];
-  const foregroundEffectOption =
-    foregroundOptions[Math.floor(Math.random() * foregroundOptions.length)];
-
-  const chosenEffects = {
-    backgroundEffect: backgroundEffectOption[1],
-    backgroundColor: backgroundColorOption[1],
-    foregroundEffect: foregroundEffectOption[1],
+  const allOutputOptions: {[key: string]: [number, string][]} = {};
+  for (const output in outputWeightsForSelectedEmojis) {
+    const weightVector = outputWeightsForSelectedEmojis[output];
+    const options = obtainOptions(weightVector, cachedWeightsMappings[output]);
+    const topOrBottomOptions =
+      quality === ChooseEffectsQuality.GOOD
+        ? options.slice(0, numRandomOptions)
+        : options.slice(-numRandomOptions);
+    allOutputOptions[output] = topOrBottomOptions;
```

Comment:
same comment as above about doing this in a reduce vs the external define + for/in.